### PR TITLE
Preverb breakdown popups, popup responsiveness

### PR DESCRIPTION
**Pull request by Madoshakalaka**
# Splendid

- preverbs get tooltips now

Screenshot from 2020-02-08 04-41-44

- popups are intelligent now

    They wrap. They adapt, they don't do crap.

Screenshot from 2020-02-08 04-41-37

@kobexamoh also I'm not trying to steal yo job. I worked on preverb breakdown and found `poppers.js` is easily adapted to all other tooltips. You can really work futher based on the current CSS!

### CODE REVIEW

**Comment by Madoshakalaka** [comment]
The coverage doesn't look good. I'll come up with tests under this PR later then.

File: CreeDictionary/API/models.py [hunk 1]
```diff
@@ -133,7 +143,7 @@ class SearchResult:
 
     # Sequence of all preverb tags, in order
     # Optional: we might not have some preverbs in our database
-    preverbs: Tuple[Tuple[str, Optional[WordformID]], ...]
+    preverbs: Tuple[Preverb, ...]
```
Comment (Madoshakalaka) [inline CreeDictionary/API/models.py]: @eddieantonio could haved passed in Wordform. But I remembered the convention that dumb objects should be passed.

File: CreeDictionary/CreeDictionary/templates/CreeDictionary/word-entries.html [hunk 2]
```diff
@@ -21,15 +21,22 @@ <h2 class="definition-title definition-title--search-result">
...
                         {% if result.linguistic_breakdown_head or result.linguistic_breakdown_tail %}
-                            <div class="definition-title__tooltip">
-                                <img class="definition-title__question-mark"
-                                     src="{% static 'CreeDictionary/images/fa/question-circle-solid.svg' %}"
-                                     alt="linguistic breakdown" data-cy="question-mark">
-                                <span class="definition-title__tooltip-text"
-                                      data-cy="linguistic-breakdown">{% if result.linguistic_breakdown_head %}
+
+
+                            <div tabindex="0" class="definition-title__tooltip-icon" aria-describedby="tooltip">
```
Comment (Madoshakalaka) [inline CreeDictionary/CreeDictionary/templates/CreeDictionary/word-entries.html]: fixes #237

File: src/css/_tooltip.css [hunk 3]
```diff
@@ -0,0 +1,55 @@
+/* adapted from https://popper.js.org/docs/v2/tutorial/ */
+
+/* how to use: Create `icon` element and `popup` element in html
+
+`popup` needs to be the direct next sibling of `icon` and has class tooltip
+
+.tooltip class in this file decides how the popup looks like
```
Comment (Madoshakalaka) [inline src/css/_tooltip.css]: @kobexamoh I think you should work on this file to improve the looks of the popup.

File: CreeDictionary/API/models.py (hunk 1 above)
Comment (eddieantonio) [reply CreeDictionary/API/models.py]: Dumb objects _only when rendering a view_! Passing in a wordform here is okay!

Comment (Madoshakalaka) [reply CreeDictionary/API/models.py]: We do use SearchResult instances directly while rendering entries

**Comment by aarppe** [comment]
@Madoshakalaka Yes, now the linguistic breakdown/analysis and preverb pop-ups are actually genuinely usable. @kobexamoh So the next step is working on making the solution more general and fully working on different screen sizes/types, as well as polishing the output so that the linguistic breakdown is visually easy to perceive. E.g. AEW has commented that having one feature relabeling per line would be better than having some relabeling split on two lines. I don't have any definitive views on this, other than perhaps making the lemma/stem easily discernible among all the feature relabelings.

**Pull request merged**